tests/test_minimal/test_tools/test_backend_and_dataset_configuration/test_zarr_dataset_configuration_model.py

```diff
@@ -0,0 +1,212 @@
+
+    expected_print = """
+acquisition/TestElectricalSeries/data
+-------------------------------------
+  dtype : int16
+  full shape of source array : (1800000, 384)
+  full size of source array : 1.38 GB
+
+  buffer shape : (1250000, 384)
+  maximum RAM usage per iteration : 0.96 GB
+
+  chunk shape : (78125, 64)
+  disk space usage per chunk : 10.00 MB
+
+  compression method : gzip
+
+  filter methods : ['blosc']
+  filter options : [{'clevel': 5}]
+
+"""
+    assert out.getvalue() == expected_print
+
+
+def test_zarr_dataset_configuration_repr():
+    """Test the programmatic repr of a ZarrDatasetConfiguration model is more dataclass-like."""
+    zarr_dataset_configuration = mock_ZarrDatasetConfiguration()
+
+    # Important to keep the `repr` unmodified for appearance inside iterables of DatasetInfo objects
+    expected_repr = (
+        "ZarrDatasetConfiguration(dataset_info=DatasetInfo(object_id='481a0860-3a0c-40ec-b931-df4a3e9b101f', "
+        "location='acquisition/TestElectricalSeries/data', dataset_name='data', dtype=dtype('int16'), "
+        "full_shape=(1800000, 384)), chunk_shape=(78125, 64), buffer_shape=(1250000, 384), compression_method='gzip', "
+        "compression_options=None, filter_methods=None, filter_options=None)"
+    )
+    assert repr(zarr_dataset_configuration) == expected_repr
+
+
+def test_validator_filter_options_has_methods():
+    with pytest.raises(ValueError) as error_info:
+        ZarrDatasetConfiguration(
+            dataset_info=mock_DatasetInfo(),
+            chunk_shape=(78_125, 64),
+            buffer_shape=(1_250_000, 384),
+            filter_methods=None,
+            filter_options=[dict(clevel=5)],
+        )
+
+    expected_error = (
+        "`filter_methods` is `None` but `filter_options` is not `None` "
+        "(received `filter_options=[{'clevel': 5}]`)! (type=value_error)"
+    )
+    assert expected_error in str(error_info.value)
+
+
+def test_validator_filter_methods_length_match_options():
+    with pytest.raises(ValueError) as error_info:
+        ZarrDatasetConfiguration(
+            dataset_info=mock_DatasetInfo(),
+            chunk_shape=(78_125, 64),
+            buffer_shape=(1_250_000, 384),
+            filter_methods=["blosc", "delta"],
+            filter_options=[dict(clevel=5)],  # Correction would be to add a second element `dict()` to avoid ambiguity
+        )
+
+    expected_error = (
+        "Length mismatch between `filter_methods` (2 methods specified) and `filter_options` (1 options found)! "
+        "`filter_methods` and `filter_options` should be the same length. (type=value_error)"
+    )
+    assert expected_error in str(error_info.value)
+
+
+def test_available_zarr_compression_methods_not_empty():
+    assert len(AVAILABLE_ZARR_COMPRESSION_METHODS) > 0
+
+
+def test_default_compression_is_always_available():
+    assert "gzip" in AVAILABLE_ZARR_COMPRESSION_METHODS
+
+
+def test_get_data_io_keyword_arguments():
```

Comment:
Note that in a future PR you change your name to `get_data_io_kwargs` which I think is a better name as referring to keyword arguments is a common python idiom and we use this in other places in the library. Let's change it now from the base so its propagated upwards.

Reply:
Fine, for the sake of getting this through I'll just change to `kwargs`